### ISSUE TITLE
implementation-dependent bit shifts

### ISSUE THREAD

**Issue by chad-iris** [iris-edu/libmseed#2]
Some of the bit shifting is done in a manner left undefined by C99, which leave it implementation-dependent.  As far as I can tell all test systems I've tried do it the same way, implying that a negative effect is not widespread.

The undefined shifts are reported by the compiler upon request:
`CFLAGS="-Wall -g -fsanitize=undefined,address"`

Reported by @gillian-iris

**Comment by chad-iris** [comment]
Potential solution to fix undefined shifts in unpackdata.c:
```C
/* Extract bit range and shift to start, consistent & defined for unsigned values */
#define EXTRACTBITRANGE(VALUE, STARTBIT, LENGTH) ((VALUE >> STARTBIT) & ((1U << LENGTH) - 1))
```

Need other fixes for packdata.c undefined shifts as reported in the test suite.

**Comment by chad-iris** [comment]
All undefined bit shift related operations have been fixed in ce5a8aaca324236a641324cf1e80dc67eb113695.
